dask/dataframe/core.py

```diff
@@ -805,4 +831,28 @@ def get(dsk, keys, get=get_sync, **kwargs):
     return get(dsk2, keys, **kwargs)  # use synchronous scheduler for now
 
 
+def pd_split(df, p, seed=0):
+    """ Split DataFrame into multiple pieces pseudorandomly
+
+    >>> df = pd.DataFrame({'a': [1, 2, 3, 4, 5, 6],
+    ...                    'b': [2, 3, 4, 5, 6, 7]})
+
+    >>> a, b = pd_split(df, [0.5, 0.5], seed=123)  # roughly 50/50 split
+    >>> a
+       a  b
+    1  2  3
+    2  3  4
+    5  6  7
+
+    >>> b
+       a  b
+    0  1  2
+    3  4  5
+    4  5  6
+    """
+    p = list(p)
+    index = pseudorandom(len(df), p, seed)
+    return [df.iloc[index == i] for i in range(len(p))]
```

Comment:
that's still not the same thing. here's an example:

let's say that `p = [0.5, 0.5]`

```
index = np.array([0, 0, 1, 0, 1, 0, 0, 1, 1, 1])
df = pd.DataFrame(np.random.randn(10, 2), columns=list('ab'))
[df.take(0), df.take(1)] != [df.iloc[index == 0], df.iloc[index == 1]]
```